src/main/kotlin/kotlinx/kover/util/Versions.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2017-2022 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.util
+
+internal class SemVer(val major: Int, val minor: Int, val patch: Int): Comparable<SemVer> {
+    companion object {
+        fun ofThreePartOrNull(version: String): SemVer? {
+            val parts = version.split(".")
+            if (parts.size != 3) return null
+
+            val major = parts[0].toIntOrNull()?: return null
+            val minor = parts[1].toIntOrNull()?: return null
+            val patch = parts[2].toIntOrNull()?: return null
```

Comment:
What about smthing like `1.2.3-SNAPSHOT`? I think it may be a case when someone wants to try locally-built agents.

Reply:
I haven't decided how to compare the `-SNAPSHOT` version and the regular one, so at the moment I decided to leave it that way.
In any case, there is now a requirement for the minimum version of the agent.

Reply:
For now, you can just use smth like `.removeSuffix("-SNAPSHOT")` or `.removeAfter("-")`